openedx/openedx-platform
# [BD-14] [WIP] [SE-2936] Modernize Video Block editor for Blockstore runtime.

This PR creates a modernized video block editor for use with the new library authoring MFE. It is currently incomplete, and is here to serve the purpose of getting arch review and feedback before continuing. It contains several comments in places where it is unclear what the best path forward is.

**Screenshots**:

As mentioned, CSS is not working, so it looks wonky:

![image](https://user-images.githubusercontent.com/1099483/94476411-10973800-0196-11eb-8b37-50bd6cc3fd74.png)

![image](https://user-images.githubusercontent.com/1099483/94476455-1ee55400-0196-11eb-80d5-e538ddddb484.png)

**Merge deadline**: None, yet.

**Testing instructions**:

1. Check this branch out, and use this frontend branch: https://github.com/edx/frontend-app-library-authoring/pull/13
2. Add a video block to a library.
3. Edit the video block.
4. Play around with the settings and save them.

**Author notes and concerns**:

1. The 'Transitional Service' will be removed, as it has been determined that although preserving `studio_view` as the editing view would be a laudable achievement and would avoid changing the API, the amount of effort involved and the number of considerations required for backwards compatibility are just too great for the case of XModules and a great deal of shimming work which need not be done with properly isolated XBlocks is being done to allow compatibility with old blocks. The new view will be `edit_view`.
2. It's not clear what the `gettext` hack included would do in a real environment or if it's even helpful.
3. edX's version of React had to be upgraded to support the features used in these new components. I've tried to avoid upgrading much else.
4. It may be possible to include the MFE libraries but I have not yet attempted it and think it's unlikely to work.
5. The version of Paragon in the platform probably can't be upgraded without changing code in several other places, though I haven't tried.
6. It's unclear how to best handle CSS for this task. Following the instructions in the README of the installed version of Paragon does not work.
7. File uploads-- transcripts and handouts-- are not yet supported. Adding them in seemed a bit premature with the number of open questions so far, since they seem to be a project in their own right.

**Reviewers**
